pkg/cvo/cvo.go

```diff
@@ -168,6 +175,11 @@ type Operator struct {
 	// moving resource, so it is not re-detected live.
 	requiredFeatureSet string
 
+	// enabledGates contains flags that control CVO functionality gated by product feature gates. The
+	// names do not correspond to product feature gates, the booleans here are "smaller" (product-level
+	// gate will enable multiple CVO behaviors).
+	enabledGates FeatureGates
```

Comment:
micro-nit: can we use `featureGates` or `enabledFeatureGates` or something as the property name?  I don't see anything about the package's `Operator` structure that would bring "oh, this property is implicitly about features" context.  It also feels like "enabled" and "gates" are sort of poking at the same idea, and maybe we don't need both.  I'd be fine with `enabledFeatures EnabledFeature` too.  Also, the way you have it will work fine, so feel free to leave it alone and ignore this comment too.

Reply:
I'll go with `enabledFeatureGates`. The problem I was trying to solve is that I find confusing that CVO now has two feature gate-related behaviors:
- enforce part of OCP feature gate architecture by filtering managed resources
- change its own behavior based on what feature gates are enabled/disabled

I tried to keep these two intents separate at least in naming and tried to avoid using the work "feature"in names related to the latter, because the code related to the former uses "feature" already. But I guess that only creates a new kind of confusion :shrug:

Reply:
Addressed in 36f6042564f499050e7f245a87c6e5ed0e34f596